### PR TITLE
feat: [IDP-1913]: Extract and report user inform in audit trails for …

### PR DESCRIPTION
**Pull request by vikyathharekal**
…backstage entities

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/57436)

### CODE REVIEW

**Review by sathish-soundarapandian — COMMENTED** [review]
> Reviewed 7 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57436)* status: 7 file(s) to review. 0 of 2 approvals obtained. 

___
*[`idp-service/contracts/openapi/v1/openapi.yaml` line 3933 at r1](https://reviewable.io/reviews/harness/harness-core/57436#-NnqL9vFAoS0skPpfc1L:-NnqL9vFAoS0skPpfc1M:bmkty9d) ([raw file](https://github.com/harness/harness-core/blob/15afaa7010d77d03a5eaafe09e321e7eb51be025/idp-service/contracts/openapi/v1/openapi.yaml#L3933)):*
> ```YAML
>             - sync
>             - async
>         user:
> ```

Should it be made as required ?

**Review by vikyathharekal — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57436)* status: 7 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`idp-service/contracts/openapi/v1/openapi.yaml` line 3933 at r1](https://reviewable.io/reviews/harness/harness-core/57436#-NnqL9vFAoS0skPpfc1L:-NnqkcJZ0vefqucCB6ac:b-tj2tc2) ([raw file](https://github.com/harness/harness-core/blob/15afaa7010d77d03a5eaafe09e321e7eb51be025/idp-service/contracts/openapi/v1/openapi.yaml#L3933)):*
<details><summary><i>Previously, sathish-soundarapandian (Sathish) wrote…</i></summary><blockquote>

Should it be made as required ?
</blockquote></details>

If APIs are hit from a script, we'll not have user info and it'll go as a system event.
This is the reason I didn't make it mandatory. 
If/when we have any workaround (we may have to derive the user from APIKey) for this, we can make it required.

**Review by devesh-harness — APPROVED** [review]

File: 953-events-api/src/main/proto/io/harness/eventsframework/schemas/idp/idp_catalog_entities_sync_capture_event.proto [hunk 1]
```diff
@@ -15,4 +15,7 @@ message IdpCatalogEntitiesSyncCaptureEvent {
   string entity_uid = 2;
   string action = 3;
   string sync_mode = 4;
+  string user_uuid = 5;
```
Comment (sathish-soundarapandian) [inline 953-events-api/src/main/proto/io/harness/eventsframework/schemas/idp/idp_catalog_entities_sync_capture_event.proto:18]: nit: We can put user related details in User object. Same convention on what we have at API side

**Review by sathish-soundarapandian — APPROVED** [review]

**Comment by vikyathharekal** [comment]
trigger uts

**Review by devesh-harness — APPROVED** [review]

**Comment by vikyathharekal** [comment]
trigger ut1

**Review by mankrit-singh — APPROVED** [review]

**Comment by vikyathharekal** [comment]
trigger ut9

**Comment by vikyathharekal** [comment]
trigger review

**Comment by vikyathharekal** [comment]
trigger uts

**Comment by vikyathharekal** [comment]
trigger checkstylecodeformat, ut8

**Comment by vikyathharekal** [comment]
trigger ut5, ut7, ut8

**Comment by vikyathharekal** [comment]
trigger uts

**Comment by vikyathharekal** [comment]
trigger copyrightcheck

**Comment by vikyathharekal** [comment]
trigger pmd

**Comment by vikyathharekal** [comment]
trigger uts

**Comment by vikyathharekal** [comment]
trigger uts

**Pull request merged**